File: gestao-spring/src/main/java/com/erp/gestao/service/MarcaService.java
package com.erp.gestao.service;

import com.erp.gestao.dto.filterDTO.MarcaFilterDTO;
import com.erp.gestao.model.Marca;
import com.erp.gestao.repository.MarcaRepository;
import com.erp.gestao.utils.BaseService;
import com.erp.gestao.utils.SpecificationBuilder;
import com.erp.gestao.utils.validate.ValidateMetodsUtils;
import org.hibernate.service.spi.ServiceException;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service()
public class MarcaService extends BaseService<Marca, Integer> {

    @Autowired
    private MarcaRepository repository;

    @Override
    public MarcaRepository getRepository() { return repository; }

    public List<Marca> listar(MarcaFilterDTO filter) {
        return repository.findAll(
                SpecificationBuilder.of(Marca.class)
                        .likeIgnoreCase("nome", filter.getNome())
        );
    }

    @Override
    protected void validate(Marca entity) throws ServiceException {

        ValidateMetodsUtils.validate(entity, genericUniqueValidator);

        super.validate(entity);
    }

}

File: gestao-angular/src/app/shared/service/marca.service.ts
import {Injectable} from "@angular/core";
import {CrudServicePadrao} from "../utils/service/crud.service.padrao";
import {HttpClient} from "@angular/common/http";
import {Marca} from "../model/marca";
import {MarcaFilterDTO} from "../dto/filterDTO/marca.filter.dto";

@Injectable({ providedIn: 'root' })
export class MarcaService extends CrudServicePadrao<Marca,MarcaFilterDTO> {

  constructor(http: HttpClient) {
    super(http, 'marca');
  }

}
